x-pack/plugins/snapshot_restore/common/lib/flatten.ts

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+export const flatten = (source: any, path: any[] = []): { [key: string]: any } => {
```

Comment:
I think a couple brief unit tests around this wouldn't hurt. At the very least it would communicate to a dev what the expected inputs and outputs are. You might want to include a test case where one of the property names contains a period, just to make it clear that the path isn't always perfectly represented by the flattened key.

Reply:
but this was written by @lukasolson, and is therefore impossible to break! just kidding 😆 I'll add some tests